src/cli/commands/add.js

```diff
@@ -204,7 +204,7 @@ export async function run(
     throw new MessageError(reporter.lang('missingAddDependencies'));
   }
 
-  const lockfile = await Lockfile.fromDirectory(config.cwd, reporter);
+  const lockfile = await Lockfile.fromDirectory(config, config.cwd, reporter);
```

Comment:
In that case would it be better to pass no the full config to the fromDirectory function?
Technically you need only the `config.worktreeFolder || config.cwd` and is `dir` needed then?